gateway/src/project.rs

```diff
@@ -300,6 +302,11 @@ where
         if let Ok(Self::Errored(errored)) = &mut new {
             errored.ctx = Some(Box::new(previous));
             error!(error = ?errored, "state for project errored");
+
+            if errored.kind == ProjectErrorKind::NoNetwork {
```

Comment:
Adding this here means that any time a project ends up in an errored state from a previous state, as long as the error kind is `::NoNetwork`, we consider that, in fact, we had no `previous -> errored` state transition, but instead, we had `previous -> new`. This also means nothing on the outside sees the transition to `-> errored` because it is swallowed by this catch all.

This is potentially not what we want, but I'm not sure. Based on the description of the solution (i.e. moving the "connect to network" outside of container creation), I think what we actually want is a state inbetween started and ready (like "attaching" state) which runs the network connect steps. And then, just like if containers disappear they are recreated, this code should be in `refresh` so that it gets triggered by the health checks.

Reply:
I ended up adding an `attaching` state between `creating` and `starting` because it produces less errors / edge cases when compared to attaching after `started`